remme/rest_api/certificate.py

```diff
@@ -14,111 +14,53 @@
 # ------------------------------------------------------------------------
 
 import re
-import datetime
 import hashlib
 from connexion import NoContent
 from cryptography.hazmat.primitives import serialization
-from cryptography.hazmat.primitives.asymmetric import rsa, padding
-from cryptography.hazmat.backends import default_backend
-from cryptography import x509
-from cryptography.x509.oid import NameOID
+from cryptography.hazmat.primitives.asymmetric import padding
 from cryptography.hazmat.primitives import hashes
 
 from remme.certificate.certificate_client import CertificateClient
+from remme.rest_api.certificate_api_decorator import certificate_put_request,\
+    http_payload_required, certificate_address_request
 from remme.shared.exceptions import KeyNotFound
 
 
-def _make_address_from_pem(client, certificate_pem):
-    certificate = x509.load_pem_x509_certificate(certificate_pem.encode('utf-8'),
-                                                 default_backend())
-    crt_bin = certificate.public_bytes(serialization.Encoding.DER).hex()
-    address = client.make_address_from_data(crt_bin)
-    return address
-
-
-def post(payload):
+@http_payload_required('certificate')
```

Comment:
Totally not needed. Connexion performs the validation of this part.

Reply:
 If you try pass empty json(payload NoneType) or json with no certificate prop(KeyError: 'certificate'), there will be 500 error, so I've added that for explicit error codes and messages binding

Reply:
This should be resolved by putting the `required` attribute into the OpenAPI specification.

Reply:
Sounds valid. Will try that, thanks